app/controllers/index.rb

```diff
@@ -0,0 +1,164 @@
+#TODO: -stretch goal- redirect back to previous page if you go to login or register. 
+
+#User.cars or User.prices gets cars or prices for that instance of user. 
+
+
+get '/' do
+	redirect '/cars'
+end
+
+get '/login' do
+	redirect '/cars' if logged_in?
+	
+	erb :login
+end
+
+#authenticates login
+post '/login' do
+	@email = params[:email]
+	@password = params[:password]
+	@user = user_at_email(params[:email])
+	
+	if fields_not_blank?( @email, @password )
+		if user_is_authenticated?( @user, @password )
```

Comment:
the method name `user_is_authenticated?` sounds like its checked to see if a user is already authenticated. As apposed to if the given email and password are valid. I think you should just drop the method for:

```ruby
if @user && @user.authenticate(@password)
```